pages/search.js

```diff
@@ -11,44 +10,73 @@ import logger from "@config/logger";
 import Input from "@components/form/Input";
 import { getTags } from "./api/discover/tags";
 import { getProfiles } from "./api/profiles";
-import config from "@config/app.json";
 import Pagination from "@components/Pagination";
 
-export async function getStaticProps() {
-  const pageConfig = config.isr.searchPage; // Fetch the specific configuration for this page
-  let data = {
-    tags: [],
-  };
-  let users = [];
-  try {
-    users = await getProfiles();
-  } catch (e) {
-    logger.error(e, "ERROR search users");
+async function fetchUsersByKeyword(keyword) {
+  const res = await fetch(
+    `${process.env.NEXT_PUBLIC_BASE_URL}/api/search?${new URLSearchParams({
+      slug: keyword,
+    }).toString()}`
+  );
+
+  const searchData = await res.json();
+  return searchData.users || [];
+}
+
+async function fetchRandomUsers() {
+  const users = await getProfiles();
+
+  if (users.length > 9) {
+    return users.sort(() => 0.5 - Math.random()).slice(0, 9);
   }
 
+  return users;
+}
+
+async function fetchTags() {
   try {
-    data.tags = await getTags();
+    return await getTags();
   } catch (e) {
     logger.error(e, "ERROR loading tags");
+    return [];
   }
+}
 
-  if (users.length > 9) {
-    data.randUsers = users.sort(() => 0.5 - Math.random()).slice(0, 9);
-  } else {
-    data.randUsers = users;
+export async function getServerSideProps(context) {
+  const { keyword } = context.query;
+
+  let serverProps = {
+    tags: [],
+    filteredUsers: [],
+    randUsers: [],
+  };
+
+  try {
+    if (keyword) {
+      serverProps.filteredUsers = await fetchUsersByKeyword(keyword);
+    }
+    else {
+      serverProps.randUsers = await fetchRandomUsers();
+    }
+  } catch (e) {
+    logger.error(e, "ERROR fetching users");
   }
 
+  serverProps.tags = await fetchTags();
```

Comment:
I think it may be better to move `fetchTags` to the try block too

Reply:
It is already under a try/catch block in the above code

Reply:
sorry, haven't noticed :)